File: java/counttozero.java
import java.util.InputMismatchException;
import java.util.Scanner;
import static java.lang.Math.abs;

public class counttozero {

    public static void ctz(int num)
    {
        int[]ctzlist = new int[abs(num)+1];  //An array for holding descending (or ascending) numbers
        int i = 0;  // Variable for iterating through array to fill it with numbers
        if (num<0)  // If user input is less than zero
        {
            while (num<1)  // loop until count reaches zero
            {
                ctzlist[i] = num; // put number in this array index
                i++;  // move up to the next array index
                num++;  // get one number closer to zero
            }
        }
        else  // If user input is greater than (or equal to) zero
        {
            while (num>-1)  // loop until count reaches zero
            {
                ctzlist[i] = num;
                i++;
                num--;
            }
        }
        for (int j : ctzlist) {  // For every index in the array
            System.out.print(j + " "); // print the number
        }
    }

    public static void main(String[] args){
        int uint;  // Variable that will hold user input
        boolean firsttime = true;
        while (true)  // Program won't leave loop until it reaches a break statement
        {
            Scanner uinput = new Scanner(System.in);  // Scanner object for reading user input
            try
            {
                if (firsttime)  // If this is your first time in the loop
                    System.out.print("Begin countdown from what number?: ");
                else
                    System.out.print("Please enter an integer: ");
                uint = uinput.nextInt();  // user input is saved in this variable
                ctz(uint); // Call method that counts to zero
                break; // escape loop if input is valid
            }
            catch (InputMismatchException e) {  // if input is invalid
                firsttime = false;  // The next time through the loop is not the first
            }
        }
    }
}

File: python/counttozero.py
def ctz(num):
    ctzlist = []  # Create a list for holding descending (or ascending) numbers
    if num == 0:  # If user enters 0
        print(0)
    elif num < 0:  # if user enters a number less than 0
        while num < 1:  # Loop until the count reaches 0
            ctzlist.append(num)  # Add the number to the list
            num = num + 1  # increase the number by one
    else:  # If user enters a number greater than 0
        while num > -1:  # Loop until the count reaches zero
            ctzlist.append(num)
            num = num - 1  # decrease the number by one
    # Print the list.  It should start with the entry and go in order until it reaches 0.
    print(*ctzlist, sep=', ')


firsttime = True
while True:  # Loop until code reaches a break statement
    try:
        if firsttime:  # If this is your first time being prompted...
            # ...prompt user to enter a number to start the countdown
            ctz(int(input("Begin countdown from what number?: ")))
            break  # Escape loop if input is valid
        else:  # If this is not your first time through the loop
            ctz(int(input("Please enter an integer: ")))  # Prompt user to enter valid input
            break  # escape loop if input is valid
    except:  # If input is invalid
        firsttime = False  # The next time through the loop is not the first time
